pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -492,7 +497,12 @@ def select_cluster_environment(self) -> ClusterEnvironment:
 
     def set_distributed_mode(self, distributed_backend: Optional[str] = None):
 
-        if distributed_backend is not None:
+        if distributed_backend is None and self.is_training_type_in_plugins:
+            return
+
+        if distributed_backend is not None and distributed_backend in TrainingTypePluginsRegistry:
+            self.distributed_backend = TrainingTypePluginsRegistry[distributed_backend]["distributed_backend"]
```

Comment:
Could you explain what's happening here @kaushikb11?

Reply:
Yup, when you go through the `set_distributed_mode` function, you will see that there are certain decisions being made based on `distributed_backend`. Hence, I am assigning `distributed_backend` from the Registry. @SeanNaren

Reply:
The flow goes like this for plugin `ddp_find_unused_parameters_false`:
handle_plugins(`ddp_find_unused_parameters_false`) => set_distributed_mode(`ddp_find_unused_parameters_false`) => set distributed_backend(`ddp`) => decisions based on `ddp`.

Reply:
Any reason we can't do `self.distributed_backend = TrainingTypePluginsRegistry[distributed_backend]` this should assign a plugin right?

Reply:
@SeanNaren 

```python
from pytorch_lightning.plugins import TrainingTypePluginsRegistry

print(TrainingTypePluginsRegistry["ddp_find_unused_parameters_false"])
```

{'description': 'DDP Plugin with `find_unused_parameters` as False', 'init_params': {'find_unused_parameters': False}, 'plugin': <class 'pytorch_lightning.plugins.training_type.ddp.DDPPlugin'>, 'distributed_backend': 'ddp'}

Reply:
Oh of course my bad :)

Given the naming and hopefully eventually cleaning up such that `distributed_backend` becomes clearer in it's responsibility, this is fine!